yb-voyager/cmd/assessMigrationCommand.go

```diff
@@ -1958,6 +2093,424 @@ func getSupportedVersionString(minimumVersionsFixedIn map[string]*ybversion.YBVe
+		var err error
+		missingPerms, pgssEnabledForAssessment, err = source.DB().GetMissingAssessMigrationPermissions()
+		if err != nil {
+			return fmt.Errorf("failed to get missing assess migration permissions: %w", err)
+		}
+		if len(missingPerms) > 0 {
+			color.Red("\nPermissions missing in the source database for assess migration:\n")
+			output := strings.Join(missingPerms, "\n")
+			utils.PrintAndLogf("%s\n\n", output)
+
+			link := "https://docs.yugabyte.com/preview/yugabyte-voyager/migrate/migrate-steps/#prepare-the-source-database"
+			fmt.Println("Check the documentation to prepare the database for migration:", color.BlueString(link))
+
+			reply := utils.AskPrompt("\nDo you want to continue anyway")
+			if !reply {
+				return fmt.Errorf("grant the required permissions and try again")
+			}
+		}
+		return nil
+	}
+
+	// PostgreSQL multi-node: check primary + all replicas
+	pg, ok := source.DB().(*srcdb.PostgreSQL)
+	if !ok {
+		return fmt.Errorf("source database is not PostgreSQL")
+	}
+
+	// Print appropriate message based on replica count
+	if len(validatedReplicaEndpoints) == 0 {
+		utils.PrintAndLogf("\nChecking permissions on database...")
+	} else {
+		utils.PrintAndLogf("\nChecking permissions on all nodes (primary + %d replica(s))...", len(validatedReplicaEndpoints))
+	}
+
+	type NodePermissionResult struct {
+		NodeName     string
+		IsPrimary    bool
+		MissingPerms []string
+		PgssEnabled  bool
+		Error        error
+	}
+
+	var results []NodePermissionResult
+
+	// Check primary
+	checkIfSchemasHaveUsagePermissions()
+	missingPerms, pgssEnabled, err := pg.GetMissingAssessMigrationPermissions()
+	if err != nil {
+		return fmt.Errorf("failed to check permissions on primary: %w", err)
+	}
+	results = append(results, NodePermissionResult{
+		NodeName:     "primary",
+		IsPrimary:    true,
+		MissingPerms: missingPerms,
+		PgssEnabled:  pgssEnabled,
+		Error:        nil,
+	})
+	pgssEnabledForAssessment = pgssEnabled // Set global flag based on primary
+
+	// Check each replica
+	for _, replica := range validatedReplicaEndpoints {
+		// Create a new Source with replica's host/port
+		replicaSource := srcdb.Source{
+			DBType:         source.DBType,
+			Host:           replica.Host,
+			Port:           replica.Port,
+			DBName:         source.DBName,
+			User:           source.User,
+			Password:       source.Password,
+			Schema:         source.Schema,
+			SSLMode:        source.SSLMode,
+			SSLCertPath:    source.SSLCertPath,
+			SSLKey:         source.SSLKey,
+			SSLRootCert:    source.SSLRootCert,
+			SSLCRL:         source.SSLCRL,
+			NumConnections: source.NumConnections,
+		}
+
+		// Create a new PostgreSQL connection for this replica
+		replicaDB := replicaSource.DB().(*srcdb.PostgreSQL)
+
+		err := replicaDB.Connect()
+		if err != nil {
+			results = append(results, NodePermissionResult{
+				NodeName:  replica.Name,
+				IsPrimary: false,
+				Error:     fmt.Errorf("failed to connect: %w", err),
```

Comment:
have a separate field in this NodePermissionResult as `ConnectionFailed boolean` to be used later in the analysis of all node results, as you can't rely on Error only for the connection failed or not.